### PR TITLE
chore: add docs about runtime upgrades

### PR DESCRIPTION
**Pull request by tomjohnburton**

### CODE REVIEW

**Review by fernandezdavid — APPROVED** [review]

File: pages/mainnet/validator-setup/managing-runtime-upgrades.mdx [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+```shell
+sudo apt update
+sudo apt --only-upgrade install chainflip-node
+sudo apt install chainflip-engine1.1
+```
+
+You should see the following logs from the engine:
+```shell
+jouanalctl -f -u chainflip-engine1.1
```
Comment (dandanlen) [inline pages/mainnet/validator-setup/managing-runtime-upgrades.mdx:37]:
```suggestion
journalctl -f -u chainflip-engine1.1
```

File: pages/mainnet/validator-setup/managing-runtime-upgrades.mdx [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+When we announce that a new version of the software is available you can already get ready by running the second version of the engine.
+
+```shell
+sudo apt update
+sudo apt --only-upgrade install chainflip-node
+sudo apt install chainflip-engine1.1
+```
+
+You should see the following logs from the engine:
```
Comment (dandanlen) [inline pages/mainnet/validator-setup/managing-runtime-upgrades.mdx:35]:
```suggestion
You should see the following logs from the new engine:
```

File: pages/mainnet/validator-setup/managing-runtime-upgrades.mdx [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+```shell
+jouanalctl -f -u chainflip-engine1.1
+{"timestamp":"2023-12-11T16:26:38.381549Z","level":"INFO","fields":{"message":"This version '1.1.0' is incompatible with the current release '1.0.0' at block: 0x4d66…fb38. WAITING for a compatible release version."},"target":"chainflip_engine::state_chain_observer::client"}
+```
+
+### After the Runtime Upgrade
+
+Check the logs of the new version of the `chainflip-engine1.1`. You should see that it started successfully.
```
Comment (dandanlen) [inline pages/mainnet/validator-setup/managing-runtime-upgrades.mdx:43]:
```suggestion
Check the logs of the new version of the `chainflip-engine1.1`. You should see that it is now processing blocks.
```

File: pages/mainnet/validator-setup/managing-runtime-upgrades.mdx [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+
+The main difficulty is having two containers that can both access the `data.db` directory. Sadly, most cloud providers
+don't support `ReadWriteMany` on volumes. So the next best thing would be to run two containers in a single pod. The issue
+here is that there is no easy way to route traffic from the `Service` to the correct pod at the moment the Runtime Upgrade
+is pushed. It would require the operator to be available at the time of the upgrade which can be difficult considering we
+are a global network.
+
+The best solution we can offer is to build a custom image wherein both versions of the `chainflip-engine` can run simultaneously.
+Docker (while also making it clear that this isn't best practise), provides several solutions on their [website](https://docs.docker.com/config/containers/multi-service_container/).
+Of the listed soltions, using [Supervisord](http://supervisord.org/) suits us the best.
```
Comment (dandanlen) [inline pages/mainnet/validator-setup/managing-runtime-upgrades.mdx:66]:
```suggestion
Of the listed solutions, using [Supervisord](http://supervisord.org/) suits us the best.
```

File: pages/mainnet/validator-setup/managing-runtime-upgrades.mdx [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+
+To build simply run:
+
+```shell
+docker buildx build -t my_repo/my_image:tag .
+```
+
+### Before the Runtime Upgrade
+
+Make sure you are running your custom image once the team has made an announcement.
```
Comment (dandanlen) [inline pages/mainnet/validator-setup/managing-runtime-upgrades.mdx:140]: Maybe mention that the new engine will register itself on-chain to signal its readiness for the upgrade.

**Review by dandanlen — COMMENTED** [review]
Looks good, some minor comments.

File: pages/mainnet/validator-setup/managing-runtime-upgrades.mdx (hunk 5 above)
Comment (tomjohnburton) [inline pages/mainnet/validator-setup/managing-runtime-upgrades.mdx]:
```suggestion
Make sure you are running your custom image once the team has made an announcement. The new engine will register itself on-chain thereby signalling its readiness for the upgrade.
```

**Pull request merged**